### PR TITLE
Add oneMKL FFT to AnyFFT wrapper for SYCL backend

### PR DESCRIPTION
**Pull request by WeiqunZhang**
Status: I did a 2d test Examples/Tests/langmuir/inputs_2d. ~~Unfortunately the results are wrong. The issue might be in the new FFT wrapper. But it could also be in somewhere else because we have not run much on Intel GPUs.~~ It works now after explicitly setting strides in the forward domain.

### CODE REVIEW

File: Source/ablastr/math/fft/WrapMklFFT.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        sycl::event r;
+        if (fft_plan.m_dir == direction::R2C) {
+            r = oneapi::mkl::dft::compute_forward(
+                *fft_plan.m_plan,
+                fft_plan.m_real_array,
+                reinterpret_cast<amrex::Real*>(fft_plan.m_complex_array));
+        } else {
+            r = oneapi::mkl::dft::compute_backward(
+                *fft_plan.m_plan,
+                reinterpret_cast<amrex::Real*>(fft_plan.m_complex_array),
```
Comment (ax3l) [inline Source/ablastr/math/fft/WrapMklFFT.cpp]: It looks to me like input & output types are distinct types in the API:
https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2024-1/compute-forward-typename-descriptor-type-typename.html

Is there a need to cast the complex pointers to `Real*`?

Comment (WeiqunZhang) [reply Source/ablastr/math/fft/WrapMklFFT.cpp]: Yes, it does not compile with `amrex::GpuComplex<Real>*`. Either `Real*` or `std::complex<Real>*` works.

**Review by ax3l — APPROVED** [review]
Yay! :tada:

Thank you!

Can you run a 1D and 3D PSATD or ES IGF test as well just to check? :)

File: Source/ablastr/math/fft/WrapMklFFT.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+namespace ablastr::math::anyfft
+{
+
+    void setup(){/*nothing to do*/}
+
+    void cleanup(){/*nothing to do*/}
+
+    FFTplan CreatePlan(const amrex::IntVect& real_size, amrex::Real * const real_array,
+                       Complex * const complex_array, const direction dir, const int dim)
```
Comment (ax3l) [inline Source/ablastr/math/fft/WrapMklFFT.cpp]: Optional AMReX-ify:
```suggestion
    void setup () {/*nothing to do*/}

    void cleanup () {/*nothing to do*/}

    FFTplan CreatePlan (const amrex::IntVect& real_size, amrex::Real * const real_array,
                        Complex * const complex_array, const direction dir, const int dim)
```

File: Source/ablastr/math/fft/WrapMklFFT.cpp [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        fft_plan.m_real_array = real_array;
+        fft_plan.m_complex_array = complex_array;
+        fft_plan.m_dir = dir;
+        fft_plan.m_dim = dim;
+        fft_plan.m_stream = amrex::Gpu::gpuStream();
+
+        return fft_plan;
+    }
+
+    void DestroyPlan(FFTplan& fft_plan)
```
Comment (ax3l) [inline Source/ablastr/math/fft/WrapMklFFT.cpp]:
```suggestion
    void DestroyPlan (FFTplan& fft_plan)
```

**Review by ax3l — APPROVED** [review]

**Pull request merged**